src/test/java/seedu/ccacommander/storage/JsonCcaCommanderStorageTest.java

```diff
@@ -0,0 +1,114 @@
+        return new JsonCcaCommanderStorage(Paths.get(filePath)).readCcaCommander(addToTestDataPathIfNotNull(filePath));
+    }
+
+    private Path addToTestDataPathIfNotNull(String prefsFileInTestDataFolder) {
+        return prefsFileInTestDataFolder != null
+                ? TEST_DATA_FOLDER.resolve(prefsFileInTestDataFolder)
+                : null;
+    }
+
+    @Test
+    public void read_missingFile_emptyResult() throws Exception {
+        assertFalse(readCcaCommander("NonExistentFile.json").isPresent());
+    }
+
+    @Test
+    public void read_notJsonFormat_exceptionThrown() {
+        assertThrows(DataLoadingException.class, () -> readCcaCommander("notJsonFormatCcaCommander.json"));
+    }
+
+    @Test
+    public void readCcaCommander_invalidMemberCcaCommander_throwDataLoadingException() {
+        assertThrows(DataLoadingException.class, () -> readCcaCommander("invalidMemberCcaCommander.json"));
+    }
+
+    @Test
+    public void readCcaCommander_invalidAndValidMemberCcaCommander_throwDataLoadingException() {
+        assertThrows(DataLoadingException.class, () -> readCcaCommander("invalidAndValidMemberCcaCommander.json"));
+    }
+
+    @Test
+    public void readAndSaveCcaCommander_allInOrder_success() throws Exception {
+        Path filePath = testFolder.resolve("TempCcaCommander.json");
+        CcaCommander original = getTypicalCcaCommander();
+        JsonCcaCommanderStorage jsonCcaCommanderStorage = new JsonCcaCommanderStorage(filePath);
+
+        // Ensure there is a file path
+        assertNotNull(jsonCcaCommanderStorage.getCcaCommanderFilePath());
+
+        // Save in new file and read back
+        jsonCcaCommanderStorage.saveCcaCommander(original, filePath);
+        ReadOnlyCcaCommander readBack = jsonCcaCommanderStorage.readCcaCommander(filePath).get();
+        assertEquals(original, new CcaCommander(readBack));
+
+        // Modify data, overwrite exiting file, and read back
+        original.createMember(HOON);
+        original.removeMember(ALICE);
+        jsonCcaCommanderStorage.saveCcaCommander(original, filePath);
+        readBack = jsonCcaCommanderStorage.readCcaCommander(filePath).get();
+        assertEquals(original, new CcaCommander(readBack));
+
+        // Save and read without specifying file path
+        original.createMember(IDA);
+        jsonCcaCommanderStorage.saveCcaCommander(original); // file path not specified
+        readBack = jsonCcaCommanderStorage.readCcaCommander().get(); // file path not specified
+        assertEquals(original, new CcaCommander(readBack));
+
+    }
+
+    @Test
+    public void saveCcaCommander_nullCcaCommander_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> saveCcaCommander(null, "SomeFile.json"));
+    }
+
+    /**
+     * Saves {@code addressBook} at the specified {@code filePath}.
+     */
+    private void saveCcaCommander(ReadOnlyCcaCommander addressBook, String filePath) {
```

Comment:
Should the param be ccaCommander?
```suggestion
    private void saveCcaCommander(ReadOnlyCcaCommander ccaCommander, String filePath) {
```

Reply:
If so the comment should be changed also

Reply:
resolved